### PR TITLE
Cleanup / type MathInput (pt 6)

### PR DESCRIPTION
**Pull request by handeyeco**
## Summary:
The last of the TS conversions

Issue: LC-740

## Test plan:

### CODE REVIEW

File: packages/math-input/src/components/__tests__/gesture-state-machine_test.ts [hunk 1]
```diff
@@ -20,10 +22,11 @@ const eventTrackers = (buffer) => {
     ];
     callbackNames.forEach((callbackName) => {
         handlers[callbackName] = function () {
+            // eslint-disable-next-line prefer-rest-params
```
Comment (handeyeco) [inline packages/math-input/src/components/__tests__/gesture-state-machine_test.ts:25]: Couldn't make heads or tails of this one.

File: packages/math-input/src/components/__tests__/two-page-keypad_test.tsx [hunk 2]
```diff
@@ -8,7 +8,6 @@ describe("<TwoPageKeyPage />", () => {
         // Arrange
         const wrapper = mount(
             <TwoPageKeypad
-                paginationEnabled={true}
```
Comment (handeyeco) [inline packages/math-input/src/components/__tests__/two-page-keypad_test.tsx:11]: This is provided by Redux, so it's erroneous to have it as a prop

File: packages/math-input/src/components/expression-keypad.tsx [hunk 3]
```diff
@@ -110,10 +110,7 @@ class ExpressionKeypad extends React.Component<Props> {
                         keyConfig={KeyConfigs.NUM_1}
                         borders={BorderStyles.BOTTOM}
                     />
-                    <ManyKeypadButton
-                        keys={extraKeys}
-                        borders={BorderStyles.NONE}
-                    />
+                    <ManyKeypadButton keys={extraKeys} />
```
Comment (handeyeco) [inline packages/math-input/src/components/expression-keypad.tsx:113]: Logic change: didn't seem to be used

File: packages/math-input/src/components/keypad-container.tsx [hunk 4]
```diff
@@ -26,7 +26,7 @@ const {row, centered, fullWidth} = Styles;
 
 type Props = {
     active?: boolean;
-    extraKeys?: Array<Key>;
+    extraKeys?: Array<Key>; // Not used?
```
Comment (jeremywiebe) [inline packages/math-input/src/components/keypad-container.tsx:29]: It appears to be passed through to the the various keypads in `renderKeypad`?  Or is it not used inside those keypads?

File: packages/math-input/src/components/many-keypad-button.tsx [hunk 5]
```diff
@@ -3,30 +3,30 @@
...
+import type {Key} from "../data/keys";
 
+type Props = {
+    keys?: Array<Key>;
+};
+
+class ManyKeypadButton extends React.Component<Props> {
     render() {
-        const {keys, ...rest} = this.props;
+        const {keys = [], ...rest} = this.props;
```
Comment (jeremywiebe) [inline packages/math-input/src/components/many-keypad-button.tsx:23]: This is probably better handled by defining a `static defaultProps = {  keys: [] }`  in this class.

File: packages/math-input/src/components/many-keypad-button.tsx [hunk 6]
```diff
@@ -3,30 +3,30 @@
...
 
         // If we have no extra symbols, render an empty button. If we have just
         // one, render a standard button. Otherwise, capture them all in a
         // single button.
         if (keys.length === 0) {
-            return <EmptyKeypadButton {...rest} />;
+            return <EmptyKeypadButton />;
```
Comment (jeremywiebe) [inline packages/math-input/src/components/many-keypad-button.tsx:29]: Is this because `EmptyKeypadButton` doesn't take any props?

**Review by jeremywiebe — APPROVED** [review]
This is shaping up nicely.  One ask before landing.  Can you make sure that all `Array<>` types on Props in this PR are changed to `ReadonlyArray<>`? 

Then 👍

File: packages/math-input/src/components/__tests__/two-page-keypad_test.tsx (hunk 2 above)
Comment (handeyeco) [reply packages/math-input/src/components/__tests__/two-page-keypad_test.tsx:11]: They fail with and without the prop. https://khanacademy.atlassian.net/browse/LC-753

File: packages/math-input/src/components/__tests__/gesture-state-machine_test.ts [hunk 7]
```diff
@@ -20,10 +22,11 @@ const eventTrackers = (buffer) => {
...
             buffer.push([callbackName, ...arguments]);
         };
     });
-    return handlers;
+    return handlers as Handlers;
```
[the remark this answers is not in the document]
Comment (handeyeco) [reply packages/math-input/src/components/__tests__/gesture-state-machine_test.ts:29]: I tried again with a few different variations and it was never happy.

File: packages/math-input/src/components/many-keypad-button.tsx (hunk 6 above)
Comment (handeyeco) [reply packages/math-input/src/components/many-keypad-button.tsx:33]: Right, with the exception of ReduxProps

File: packages/math-input/src/components/many-keypad-button.tsx (hunk 5 above)
Comment (handeyeco) [reply packages/math-input/src/components/many-keypad-button.tsx]: Fixed

**Pull request merged**